cli.py

```diff
@@ -0,0 +1,166 @@
+        if not infile:                                                  #if filename is not provided, read string from stdin
+            input_string = input("Enter the input string:")
+            if not input_string and mode == "enc":
+                print("Null String Entered")
+            else:
+                while not input_string and mode == "dec":
+                    print("Empty String Cannot be decoded")
+                    input_string = input("Enter the input string:")
+
+            if input_format == "bin":
+                #input_string = bytes(input_string, 'utf-8')
+                input_string = input_string.encode()
+
+        else:                                                          #if file name is provided read string from file
+            while not bech32m.read_file(infile, input_format):
+                infile = input("Enter input file name:")
+
+            input_string = bech32m.read_file(infile, input_format)
+
+        #validate input string with the input format provided, if string not in format ask for input string again
+        if input_format == "b64":
+            while not bech32m.check_if_str_is_base64(input_string):
+                input_string = input("Enter Valid \"base64\" input string:")
+
+        if input_format == "hex":
+            while not bech32m.check_if_str_is_hex(input_string):
+                input_string = input("Enter valid \"Hex\" input string:")
+
+        if input_format == "bin":
+            while not bech32m.check_if_str_is_bin(input_string):
+                input_string_s = input("Enter valid \"bin\" input string:")
+                #input_string = bytes(input_string_s, 'utf-8')
+                input_string = input_string_s.encode()
+
+        #input output format and validate
+        output_format = input("Choose output Format[b64/bin/hex]"
+                              " (Press Enter for Default[\"bech32m\" for \"enc\" and \"hex\" for \"dec\"]):")
+        while output_format not in ["b64", "bin", "hex", ""]:
+            output_format = input("Enter Valid Output Format[b64/bin/hex]"
+                                  "(Press Enter for Default[\"hex\" for \"enc\" and \"bech32m\" for \"dec\"]):")
+
+        #set default output format for enc/dec if not provided
+        if mode == "enc" and output_format == "":
+            output_format = "bech32m"
+        if mode == "dec" and output_format == "":
+            output_format = "hex"
+
+        #input output file name if enc/dec data is to be written to a file
+        outfile = input("Enter output file name (press enter to print output to \"stdout\") : ")
+
+        return mode, hrp, input_string, infile, input_format, output_format, outfile
+
+
+def main():
+    mode, hrp, input_string, input_file, input_format, output_format, output_file = user_interface()
+
+    if input_string is None:
+        if not bech32m.read_file(input_file, input_format):
```

Comment:
issue (blocking): This call crashes application if non utf-8 character is in a file.